feat: Add methods to store & restore UI component state (#3208)
Pycord-Development/pycord: CHANGELOG.md

### Added

> - Methods to store and restore states of UI components.

Reviewer: ```suggestion
- Added `View.to_component_instances()`/`from_components()` for snapshotting and
  restoring UI component state.
```
